tour_offers/src/main/kotlin/pl/edu/pg/rsww/touroffers/TourOffersManager.kt

```diff
@@ -0,0 +1,126 @@
+
+    /*fun GetAvailable(db: MongoDatabase): List<String> {
+        val projection = Projection.fields(
+                Projections.include(Entity::entity_id.name),
+                Projection.excludeId()
+        )
+
+        return db.getCollection<Entity>("snapshots").find(Filters.eq(Entity::entity_type.name, "Hotel"))
+                .toList()
+                .filter{
+                    it.data?.getInteger("reservation_count") <= it.data?.getInteger("reservation_limit") ?: false
+                }
+    }*/
+
+    fun getTourList(destination: String,
+                    from: String,
+                    departureDate: String,
+                    adults: Int,
+                    under3: Int,
+                    under10: Int,
+                    under18: Int): String{
+
+        val client = MongoClient.create(connectionString = connectionString)
+        val db = client.getDatabase(databaseName = dbName)
+
+        var minAge: Int = 18
+        val numPeople: Int = adults + under3 + under10 + under18
+        if (under3 != 0){
+            minAge = 0
+        }else if(under10 != 0){
+            minAge = 3
+        }else if(under18 != 0){
+            minAge = 10
+        }
+
+        var result = ""
+
+        val destCities = db.getCollection<Entity>("snapshots").find(Filters.eq(Entity::entity_type.name, "Tour"))
+                .toList()
+                .filter{
+                    (if(destination != "") (it.data?.getString("hotel_destination_city_title")?.contains(destination) ?: false) ||
+                            (it.data?.getString("hotel_destination_country_title")?.contains(destination) ?: false) else true)
+                    && (if(from != "") it.data?.getList("hotel_from", String::class.javaObjectType)?.any{ it.contains(from) } ?: false else true)
+                            && ((it.data?.getDouble("hotel_minimum_age") ?: 18.0) <= minAge)
+                     && ((it.data?.getDouble("hotel_max_people_per_reservation")?: 0.0) >= numPeople )
+                    && (if(departureDate != "") it.data?.getString("start_date") == departureDate ?: false else true)
+                }
+
+        if (destCities != null) {
+            var n = 0
+            for (d in destCities) {
+                    result += """
+                            <div class="my-3 rounded-md outline-1 box-border border-2 shadow-md flex justify-between gap-x-6 py-5 flex min-w-0 gap-x-4 space-x-4 px-5">
+                                <div>
+                                    <p class="break-afer-auto text-sm font-semibold leading-6 text-gray-900">
+                                        ${d.data?.getString("title") ?: "Coś poszło nie tak..."}</p>
+                                    <p class="mt-1 truncate text-xs leading-5 text-gray-500">
+                                        ${d.data?.getString("hotel_destination_city_title") + "," ?: ""} ${d.data?.getString("hotel_destination_country_title")}</p>

```

Comment:
`?:` will always return left side since for an empty string you still add "," making it a truth value.